.github/workflows/pr.yml

```
@@ -137,6 +137,44 @@ jobs:
         env:
           PY_COLORS: '1'
 
+      - name: Idle Galaxy Deployment
+        run: |
+          echo "idling galaxy deployment"
+          kubectl -n galaxy patch galaxy example-galaxy --type merge -p '{"spec":{"idle_deployment":true}}'
+        shell: bash
+
+      - name: Poll Deployments and StatefulSets until all are scaled down
+        run: |
+          echo "Waiting for all deployments and statefulsets to be scaled down, excluding galaxy-operator-controller-manager"
+          while true; do
+          DEPLOYMENTS=$(kubectl -n galaxy get deployments -o json | jq '[.items[] | select(.metadata.name != "galaxy-operator-controller-manager")] | length')
+          STATEFULSETS=$(kubectl -n galaxy get statefulsets -o json | jq '.items | length')
+          if [[ "$DEPLOYMENTS" == "0" && "$STATEFULSETS" == "0" ]]; then
+            break
+          fi
+          sleep 5
+            done
+        shell: bash
+
+      - name: Un-idle Galaxy Deployment
+        run: |
+          echo "un-idling galaxy deployment"
+          kubectl -n galaxy patch galaxy example-galaxy --type merge -p '{"spec":{"idle_deployment":false}}'
+        shell: bash
+
+      - name: Poll Deployments and StatefulSets until all are scaled up
+        run: |
+          echo "Waiting for all deployments and statefulsets to be scaled up, excluding galaxy-operator-controller-manager"
+          while true; do
+          DEPLOYMENTS=$(kubectl -n galaxy get deployments -o json | jq '[.items[] | select(.metadata.name != "galaxy-operator-controller-manager")] | length')
+          STATEFULSETS=$(kubectl -n galaxy get statefulsets -o json | jq '.items | length')
+          if [[ "$DEPLOYMENTS" == "2" && "$STATEFULSETS" == "1" ]]; then
```

Shouldn't we have 5 deployments and not 2 ? (api, content, web, worker and redis)

```suggestion
          if [[ "$DEPLOYMENTS" == "5" && "$STATEFULSETS" == "1" ]]; then
```

---

"$STATEFULSETS" == "1" since we have a scenario with external database then we need a way to conditionally set 0 and 1 here

---

That makes sense, I hadn't thought of that.  I'll set a variable for expected statefulSets and add a conditional, thx.